### PR TITLE
[Core] Missing copy in pytthon for matrix and vector

### PR DESCRIPTION
**Pull request by loumalouomega**
Just that

### CODE REVIEW

File: kratos/python/add_matrix_to_python.cpp [hunk 1]
```diff
@@ -68,6 +64,7 @@ namespace Python
         binder.def("__itruediv__", [](TMatrixType& m1, const typename TMatrixType::value_type& value){ m1/=value; return m1;}, py::is_operator());
 
         binder.def("__str__", PrintObject<TMatrixType>);
+        binder.def("copy", [](const TMatrixType& self){TMatrixType aux(self.size1(), self.size2()); for (std::size_t i = 0; i < self.size1(); ++i) {for (std::size_t j = 0; j < self.size2(); ++j) {aux(i,j) = self(i,j);}}; return aux;} );
```
Comment (philbucher) [inline kratos/python/add_matrix_to_python.cpp]: Can you please shift it up by one line such that the __str__ method is the lowest one?

Other than that can you please add a test?

Comment (loumalouomega) [reply kratos/python/add_matrix_to_python.cpp]: Done

File: kratos/tests/test_matrix_interface.py [hunk 2]
```diff
@@ -1,51 +1,70 @@
...
+        A2 = A.copy()
+        b2 = b.copy()
+
+        self.assertEqual(2,A2.Size1())
+        self.assertEqual(3,A2.Size2())
+
+        for j in range(A2.Size2()):
+            self.assertEqual(b2[j], b2[j])
+            for i in range(A2.Size1()):
+                self.assertEqual(A[i,j], A2[i,j])
```
Comment (philbucher) [inline kratos/tests/test_matrix_interface.py:27]: could you please also check that the two objects are not the same one by checking the if the content is different after modifying one of them?
Thus you would make sure that not the same memory is referenced

**Review by philbucher — COMMENTED** [review]
I thought some more and I think instead of having an auxiliary method that does the copy it would be better to directly expose the Copy-Constructor.
- Then the behavior / usage is the same in python and C++
- no reimplementation necessary
- this is already done for the [Compressed Matrix](https://github.com/KratosMultiphysics/Kratos/blob/2ed201c7503f1a3827482d5621360d7ca4d41e85/kratos/python/add_matrix_to_python.cpp#L121) and e.g. the Properties => #5896

Sorry I didn't think abt this earlier, but the changes should be minor. In #5896 we basically reached the same conclusion

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**